File: bindings/lua/Polycode/LuaScriptInstance.lua
require "Polycode/ScriptInstance"

class "LuaScriptInstance" (ScriptInstance)



function LuaScriptInstance:__getvar(name)
	if name == "tableRef" then
		return Polycore.LuaScriptInstance_get_tableRef(self.__ptr)
	end
	if ScriptInstance["__getvar"] ~= nil then
		return ScriptInstance.__getvar(self, name)
	end
end


function LuaScriptInstance:__setvar(name,value)
	if name == "tableRef" then
		Polycore.LuaScriptInstance_set_tableRef(self.__ptr, value)
		return true
	end
	if ScriptInstance["__setvar"] ~= nil then
		return ScriptInstance.__setvar(self, name, value)
	else
		return false
	end
end


function LuaScriptInstance:__delete()
	if self then Polycore.delete_LuaScriptInstance(self.__ptr) end
end

File: bindings/javascript/Polycode/LuaScriptInstance.js
function LuaScriptInstance() {
	Object.defineProperties(this, {
		'tableRef': { enumerable: true, configurable: true, get: LuaScriptInstance.prototype.__get_tableRef, set: LuaScriptInstance.prototype.__set_tableRef}
	})
}
LuaScriptInstance.prototype.__get_tableRef = function() {
	return Polycode.LuaScriptInstance__get_tableRef(this.__ptr)
}

LuaScriptInstance.prototype.__set_tableRef = function(val) {
	Polycode.LuaScriptInstance__set_tableRef(this.__ptr, val)
}

Duktape.fin(LuaScriptInstance.prototype, function (x) {
	if (x === LuaScriptInstance.prototype) {
		return;
	}
	Polycode.LuaScriptInstance__delete(x.__ptr)
})
